internal/handlers/order.go

```diff
@@ -0,0 +1,108 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/AleksandrGurkin/gophermart/internal/storage/models"
+	"github.com/AleksandrGurkin/gophermart/internal/storage/orders"
+	"github.com/AleksandrGurkin/gophermart/internal/storage/sessions"
+	"github.com/AleksandrGurkin/xlog/xctx"
+	"github.com/theplant/luhn"
+)
+
+type OrderHandler struct {
+	Orders   orders.Storage
+	Sessions sessions.Storage
+}
+
+func (h *OrderHandler) LoadOrder(w http.ResponseWriter, r *http.Request) {
+	session := findToken(r.Cookies())
+	user, err := h.Sessions.GetUser(session)
+	if err != nil {
+		xctx.XLogFromCtx(r.Context()).Error(err.Error())
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	b, err := io.ReadAll(r.Body)
+	if err != nil {
+		xctx.XLogFromCtx(r.Context()).Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	id, err := readOrderNum(string(b))
+	if err != nil {
+		xctx.XLogFromCtx(r.Context()).Error(err.Error())
+		http.Error(w, err.Error(), http.StatusUnprocessableEntity)
+		return
+	}
+
+	err = h.Orders.CreateOrder(context.Background(), id, user)
+	if err != nil {
+		xctx.XLogFromCtx(r.Context()).Error(err.Error())
+		switch {
+		case errors.Is(err, orders.ErrorOrderIDAlreadyExist):
+			http.Error(w, err.Error(), http.StatusOK)
+			return
+		case errors.Is(err, orders.ErrorOrderIDAlreadyExistAnotherUser):
+			http.Error(w, err.Error(), http.StatusConflict)
+			return
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+	}
+	w.WriteHeader(http.StatusAccepted)
+}
+
+func (h *OrderHandler) ListOrders(w http.ResponseWriter, r *http.Request) {
+	session := findToken(r.Cookies())
+	user, err := h.Sessions.GetUser(session)
+	if err != nil {
+		xctx.XLogFromCtx(r.Context()).Error(err.Error())
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+	list, err := h.Orders.ListOrders(context.Background(), user)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if len(list) == 0 {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	res := ListResponse(list)
+	rb, err := json.Marshal(res)
+	if err != nil {
+		xctx.XLogFromCtx(r.Context()).Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	_, _ = w.Write(rb)
+}
+
+func readOrderNum(num string) (models.OrderID, error) {
+	id, err := strconv.ParseUint(num, 10, 64)
+	if err != nil {
+		return "bad argument", err
```

Comment:
```suggestion
		return "", fmt.Errorf("bad argument: %w", err)
```
возвращать какую-то строку не имеет смысла, так как ты не обрабатываешь ее при ошибке (а если бы обрабатывал, то это было бы довольно костыльно). В таком случае, если хочется донести этот текст до кого-нибудь, можно обернуть ошибку